libpldm/pdr.c

```diff
@@ -1126,17 +1138,17 @@ void pldm_entity_association_tree_delete_node(
 			entity.entity_instance_num &&
 		    current_entity.entity_container_id ==
 			entity.entity_container_id) {
+			if (start == parent->first_child) {
+				parent->first_child = start->next_sibling;
+			} else {
+				prev->next_sibling = start->next_sibling;
+			}
+			start->next_sibling = NULL;
 			break;
 		}
 		prev = start;
 		start = start->next_sibling;
 	}
-	if (start == parent->first_child) {
```

Comment:
Can you please post what error message we see here and are trying to solve

Reply:
This is the error msg : 
libpldm/pdr.c:1139:2: warning: Either the condition 'start!=NULL' is redundant or there is possible null pointer dereference: start. [nullPointerRedundantCheck]
 start->next_sibling = NULL;